src/test/mod.rs

```diff
@@ -0,0 +1,255 @@
+/*
```

Comment:
Since you're writing all these nice tests anyway, lets wrap them in a `Runtime` and attach a trace monitor to each test for extra testing.  See `circuit_builder.rs` for examples.

Reply:
you didn't do this, did you?